compiler/rustc_symbol_mangling/src/v0.rs

```diff
@@ -287,12 +287,37 @@ impl Printer<'tcx> for SymbolMangler<'tcx> {
             }
         }
 
-        self.path_append_impl(
-            |cx| cx.print_def_path(parent_def_id, &[]),
-            &key.disambiguated_data,
-            self_ty,
-            impl_trait_ref,
-        )
+        self.push(match impl_trait_ref {
+            Some(_) => "X",
+            None => "M",
+        });
+
+        // Skip encoding impl generic params if the substitutions contain parameters
+        // (polymorphization is enabled) and this isn't an inherent impl.
```

Comment:
Need to adjust this comment so - currently doesn't skip when this comment says it will.